mmcv/runner/hooks/checkpoint.py

```diff
@@ -109,12 +112,17 @@ def after_train_iter(self, runner):
 
         # save checkpoint for following cases:
         # 1. every ``self.interval`` iterations
-        # 2. reach the last iteration of training
-        if self.every_n_iters(
-                runner, self.interval) or (self.save_last
-                                           and self.is_last_iter(runner)):
+        # 2. reach the last `self.num_last_ckpts` iterations of training
+        if self.every_n_iters(runner, self.interval) or \
+                (self.save_last and self.near_last_iter(runner)):
             runner.logger.info(
                 f'Saving checkpoint at {runner.iter + 1} iterations')
             if self.sync_buffer:
                 allreduce_params(runner.model.buffers())
             self._save_checkpoint(runner)
+
+    def near_last_epoch(self, runner):
+        return runner.epoch + self.num_last_ckpts >= runner._max_epochs
+
+    def near_last_iter(self, runner):
```

Comment:
In iter-based runner, usually, the users may want to save checkpoint more frequently. However, save checkpoint per iteration is too frequent. A parameter interval for this case could be considered.

Reply:
So maybe we need a piece-wise interval parameter?

Reply:
We can do that. Suggest having a design first.